examples/siro_sandbox/app_states/app_state_fetch.py

```diff
@@ -0,0 +1,279 @@
+            self._sandbox_service, self.get_gui_controlled_agent_index()
+        )
+
+    def on_environment_reset(self, episode_recorder_dict):
+        self._held_target_obj_idx = None
+
+        sim = self.get_sim()
+        temp_ids, _ = sim.get_targets()
+        self._target_obj_ids = [
+            sim._scene_obj_ids[temp_id] for temp_id in temp_ids
+        ]
+
+        self._nav_helper.on_environment_reset()
+
+        self._camera_helper.update(self._get_camera_lookat_pos(), dt=0)
+
+    def get_sim(self):
+        return self._sandbox_service.sim
+
+    def _update_grasping_and_set_act_hints(self):
+        drop_pos = None
+        grasp_object_id = None
+
+        if self._held_target_obj_idx is not None:
+            if self._sandbox_service.gui_input.get_key_down(
+                GuiInput.KeyNS.SPACE
+            ):
+                # temp: drop object right where it is
+                drop_pos = self._get_target_object_position(
+                    self._held_target_obj_idx
+                )
+                drop_pos.y = 0.0
+                self._held_target_obj_idx = None
+
+            # todo: implement throwing, including viz
+        else:
+            # check for new grasp and call gui_agent_ctrl.set_act_hints
+            if self._held_target_obj_idx is None:
+                assert not self.gui_agent_ctrl.is_grasped
+                # pick up an object
+                if self._sandbox_service.gui_input.get_key_down(
+                    GuiInput.KeyNS.SPACE
+                ):
+                    translation = self._get_agent_translation()
+
+                    min_dist = self._can_grasp_place_threshold
+                    min_i = None
+                    for i in range(len(self._target_obj_ids)):
+                        this_target_pos = self._get_target_object_position(i)
+                        # compute distance in xz plane
+                        offset = this_target_pos - translation
+                        offset.y = 0
+                        dist_xz = offset.length()
+                        if dist_xz < min_dist:
+                            min_dist = dist_xz
+                            min_i = i
+
+                    if min_i is not None:
+                        self._held_target_obj_idx = min_i
+                        grasp_object_id = self._target_obj_ids[
+                            self._held_target_obj_idx
+                        ]
+
+        walk_dir = None
+        if not self._first_person_mode:
+            candidate_walk_dir = (
+                self._nav_helper.viz_and_get_humanoid_walk_dir()
+            )
+            if self._sandbox_service.gui_input.get_mouse_button(
+                GuiInput.MouseNS.RIGHT
+            ):
+                walk_dir = candidate_walk_dir
+
+        self.gui_agent_ctrl.set_act_hints(
+            walk_dir,
+            grasp_object_id,
+            drop_pos,
+            self._camera_helper.lookat_offset_yaw,
+        )
+
+        return drop_pos
```

Comment:
We don't need to return `drop_pos`, the object is dropped at `drop_pos` in `self.gui_agent_ctrl.set_act_hints`.